kernel/host/xmicterm/host/xmicterm.cc

```diff
@@ -129,16 +144,73 @@ class file_logger {
     std::unordered_map<uint16_t, std::unique_ptr<std::ofstream>> files;
 };
 
+/// Wrapper to the micctrl application, that controls the xeon phi coprocessor
+class mic_ctrl {
+public:
+    mic_ctrl(int adapter_)
+        :adapter(adapter_) {
+      }
+
+
+    void boot() {
+      std::stringstream ss;
+      ss << "sudo sh -c \"echo \\\"boot:elf:`pwd`/boot64.elf\\\" > /sys/class/mic/mic"
+          << adapter << "/state\"";
+      system(ss.str().c_str());
+    }
+
+    void reset_wait () {
+      std::stringstream ss;
+      ss << "sudo micctrl -r --wait mic" << adapter << " mic" << adapter;
+      system(ss.str().c_str());
+    }
+
+    void reset() {
+      std::stringstream ss;
+      ss << "sudo micctrl -r mic" << adapter;
+      system(ss.str().c_str());
+    }
+
+    void status() {
+      std::stringstream ss;
+      ss << "sudo micctrl -s mic" << adapter;
+      system(ss.str().c_str());
+    }
+
+private:
+    int adapter;
+};
+
+void signal_handler(int sig_type) {
+  if (sig_type == SIGINT) {
+    restore_root();
+    ctrl->reset();
+    ctrl->status();
+  }
+  exit(0);
```

Comment:
is it necessary to flush/close all log files before exit() in order to not loose some messages?